tst/testinstall/MatrixObj/testmatobj_new.g

```diff
@@ -0,0 +1,299 @@
+
+    if nrRows <> fail then
+        if nrRows <> ex.nrRows then
+            MatObjTest_HandleErrorWrongResult("NrRows", [ex.matObj], opt.breakOnError, ex, errors);
+        fi;
+    fi; 
+end;
+
+MatObjTest_TestNrCols := function(ex, opt, errors)
+    local nrCols;
+
+    nrCols := MatObjTest_CallFunc(NrCols, [ex.matObj], opt.breakOnError, errors, ex);
+    
+    if nrCols <> fail then
+        if nrCols <> ex.nrCols then
+            MatObjTest_HandleErrorWrongResult("NrCols", [ex.matObj], opt.breakOnError, ex, errors);
+        fi;
+    fi; 
+end;
+
+MatObjTest_TestMatElm := function(ex, opt, errors)
+    local col, row, elm;
+
+    col := ex.nrCols;
+    row := ex.nrRows;
+    elm := MatObjTest_CallFunc(MatElm, [ex.matObj, row, col], opt.breakOnError, errors, ex);
+
+    if elm <> fail then
+        if elm <> ex.sourceOfTruth[row,col] then
+            MatObjTest_HandleErrorWrongResult("MatElm", [ex.matObj, row, col], opt.breakOnError, ex, errors);
+        fi;
+    fi;
+end;
+
+MatObjTest_TestSetMatElm := function(ex, opt, errors)
+    local col, row, elm;
+
+    col := ex.nrCols;
+    row := ex.nrRows;
+    elm := Zero(ex.baseDomain);
+    ex.sourceOfTruth[row, col] := elm;
+    ex.mat[row, col] := elm;
+    MatObjTest_CallFunc(SetMatElm, [ex.matObj, row, col, elm], opt.breakOnError, errors, ex);
+
+    if ex.matObj[row, col] <> ex.sourceOfTruth[row, col] then
+        MatObjTest_HandleErrorWrongResult("SetMatElm", [ex.matObj, row, col, elm], opt.breakOnError, ex, errors);
+    fi;
+end;
+
+MatObjTest_TestConstructingFilter := function(ex, opt, errors)
+    local filter;
+    #Error("before CallFunc");
+    filter := MatObjTest_CallFunc(ConstructingFilter, [ex.matObj], opt.breakOnError, errors, ex);
+    #Error("made it past CallFunc");
+    if filter <> fail then
+        if filter <> ex.filter then
+            MatObjTest_HandleErrorWrongResult("ConstructingFilter", [ex.matObj], opt.breakOnError, ex, errors);
+        fi;
+    fi;
+end;
+
+
+TestMatrixObj := function(filter, opt)
+    local errors, examples, ex;
+
+    errors := [];
+
+    if not IsBound(opt.randomSource) then
+        opt.randomSource := GlobalMersenneTwister;
+    fi;
+
+    if not IsBound(opt.state) then 
+        opt.state := State(GlobalMersenneTwister);
+    fi;
+
+    if not IsBound(opt.forbidInv) then
+        opt.forbidInv := false;
+    fi;
+
+    if not IsBound(opt.breakOnError) then
+        opt.breakOnError := false;
+    fi;
+
+    examples := MatObjTest_GenerateExample(filter, opt);
+
+    for ex in examples do
+        MatObjTest_TestBaseDomain(ex, opt, errors);
+    od;
+
+    for ex in examples do
+        MatObjTest_TestNrRows(ex, opt, errors);
+    od; 
+
+    for ex in examples do
+        MatObjTest_TestNrCols(ex, opt, errors);
+    od; 
+
+    for ex in examples do
+        MatObjTest_TestMatElm(ex, opt, errors);
+    od; 
+
+    for ex in examples do
+        MatObjTest_TestSetMatElm(ex, opt, errors);
```

Comment:
This tests only works for MatrixObj types that allow modification. If the matrices are immutable, we should not try this resp. should not let this trigger an error